### PR TITLE
Bug fix/cinfra fix unstable leader election tests

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

*TestOnly: Trigger Leader in a transaction where we erase the leader and increase the plan. Any one after the other state is undefined in Supervision.*

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/js/server/replication2/replication2-replicated-state-document-store.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ const lp = require("@arangodb/testutils/replicated-logs-predicates");
 const lhttp = require('@arangodb/testutils/replicated-logs-http-helper');
 const dh = require("@arangodb/testutils/document-state-helper");
 const ch = require("@arangodb/testutils/collection-groups-helper");
+const serverHelper = require('@arangodb/test-helper');
```
Comment (apetenchea) [inline tests/js/server/replication2/replication2-replicated-state-document-store.js:38]: This is already included in line 32 as `helper`.

File: js/common/modules/@arangodb/testutils/replicated-logs-helper.js [hunk 2]
```diff
@@ -216,8 +216,16 @@ const replicatedLogSetPlanTerm = function (database, logId, term) {
 };
 
 const triggerLeaderElection = function (database, logId) {
-  serverHelper.agency.increaseVersion(`Plan/ReplicatedLogs/${database}/${logId}/currentTerm/term`);
-  serverHelper.agency.remove(`Plan/ReplicatedLogs/${database}/${logId}/currentTerm/leader`);
+  // This operation has to be in one envelope. Otherwise we violate the assumption
+  // that they are nly modified as a unit.
```
Comment (apetenchea) [inline js/common/modules/@arangodb/testutils/replicated-logs-helper.js:220]:
```suggestion
  // that they are only modified as a unit.
```

**Review by apetenchea — APPROVED** [review]
LGTM

Comments refer just to minor adjustments

File: tests/js/server/replication2/replication2-replicated-state-document-store.js (hunk 1 above)
Comment (mchacki) [reply tests/js/server/replication2/replication2-replicated-state-document-store.js]: Wasn't needed anyhow, erased

**Comment by mchacki** [comment]
Red test cannot be related to this change

**Pull request merged**